foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/OrphedomosToFabric8Migration.java

```diff
@@ -0,0 +1,277 @@
+            for (Plugin plugin : plugins) {
+                if ((pluginName).equalsIgnoreCase(plugin.getArtifactId())) {
+                    return plugin;
+                }
+            }
+        }
+        return null;
+    }
+
+    private boolean isRootProjectPom(Model model) {
+        return "com.boozallen.aissemble".equals(model.getParent().getGroupId()) && "build-parent".equals(model.getParent().getArtifactId());
+    }
+    private boolean isTestDockerPom(Model model){
+        return model.getParent().getArtifactId().endsWith("-tests") && model.getArtifactId().endsWith("-tests-docker");
+    }
+
+    private Xpp3Dom getConfiguration(Plugin plugin) {
+        Object pluginConfiguration = plugin.getConfiguration();
+        return (Xpp3Dom) pluginConfiguration;
+    }
+
+    private static boolean hasConfigurationItem(Xpp3Dom configuration, String configurationItem){
+        return configuration.getChildren(configurationItem).length > 0;
+    }
+
+    /**
+     * Function to validate whether migration should be performed or not.
+     * @param file file to validate
+     * @return true if migration should be performed
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+        boolean migrateConfiguration = false;
+        boolean migratePackaging = false;
+
+        Model model = getLocationAnnotatedModel(file);
+        if (isRootProjectPom(model) || isTestDockerPom(model)){
+            shouldExecute = false;
+        } else {
+            Plugin orphedomosPlugin = getPlugin(model, OLD_ARTIFACT_ID);
+            if (orphedomosPlugin != null){
+                migrateConfiguration = true;
+            }
+            if(containsOrphedomosPackaging(file)){
+                migratePackaging = true;
+            }
+            shouldExecute = migrateConfiguration || migratePackaging;
+        }
+
+        return shouldExecute;
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true
+     * @param file file to migrate
+     * @return isSuccessful - whether file was migrated successfully or not
+     */
+    @Override
+    protected boolean performMigration(File file) {
+        Model model = getLocationAnnotatedModel(file);
+        Plugin plugin  = getPlugin(model, OLD_ARTIFACT_ID);
+        boolean replacedConfig = false;
+        boolean replacedPackaging = false;
+        boolean isSuccessful = false;
+
+        Xpp3Dom config;
+
+        if (plugin != null){
+            config = getConfiguration(plugin);
+        } else {
+            config = null;
+            logger.info("Orphedomos plugin configuration not found. Skipping configuration migration.");
+        }
+
+        // check if pom file has orphedomos-maven-plugin configuration
+        if (config != null){
+            // grab the old imageVersion and imageName
+            IMAGE_VERSION = getImageVersion(config);
+            IMAGE_NAME = getImageName(config);
```

Comment:
A: if these aren't final strings, let's just leave them as regular variables without the all-caps (I believe that's the java convention)

Reply:
updated to `imageName` and `imageVersion`